### DISCUSSION TITLE
Support pack module

### DISCUSSION THREAD

**Discussion by Psijic** [mapsforge/vtm#1062] [Q&A] [answered]
```
GeojsonTileSource
NextzenGeojsonTileSource
TileDecoder
OkHttpEngine
```
These files I copy-pasted from vtm-parent sample to my project to implement Nextzen vector maps usage. I also implemented 
`api ("com.fasterxml.jackson.core:jackson-core:2.14.2")` 
to use it (while I use Kotlin serialization by default). Can this extras be implemented as a separate library (like osmdroid-bonuspack) at least, to easily update with the next release? Also can I easily switch `jackson` to `Kotlin serialization` and `OkHttpFactory` to `Ktor`?

**Comment by devemux86** [comment, marked as answer]
See the [integration guide](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#mapbox-vector-tiles) for the dependencies for vector tiles.

e.g. NextzenMvtTileSource uses a tile decoder for Mapbox vector tiles in vtm-mvt module

**Reply by Psijic** [reply]
Didn't find it in the library pack, copied from samples. Last time it was said here: it has too much dependencies to copy in the main lib (or something like that).
I already implemented:
```
    runtimeOnly ("org.mapsforge:vtm-android:$vtmVersion:natives-armeabi-v7a")
    runtimeOnly ("org.mapsforge:vtm-android:$vtmVersion:natives-arm64-v8a")
    runtimeOnly ("org.mapsforge:vtm-android:$vtmVersion:natives-x86")
    runtimeOnly ("org.mapsforge:vtm-android:$vtmVersion:natives-x86_64")
    implementation ("org.mapsforge:vtm-android:$vtmVersion")
    implementation ("org.mapsforge:vtm-themes:$vtmVersion")
    implementation ("com.caverock:androidsvg:1.4")
```
What should I also implement in gradle to use `NextzenGeojsonTileSource` out of the box?
[ vtm-mvt](https://mvnrepository.com/artifact/org.mapsforge/vtm-mvt)
P.S. NextzenMvtTileSource has a render bug: wrong tile positions (shuffled) on the level 2 zoom.

**Reply by devemux86** [reply]
> NextzenGeojsonTileSource

See the relevant section in the [integration guide](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#geojson-vector-tiles).
(NextzenGeojsonTileSource is in vtm-json)
